tests/model/write_model_test.go

```diff
@@ -0,0 +1,427 @@
+package model
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	parser "github.com/craigpastro/openfga-dsl-parser/v2"
+	"github.com/openfga/openfga/cmd"
+	"github.com/openfga/openfga/pkg/typesystem"
+	"github.com/stretchr/testify/require"
+	pb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	healthv1pb "google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/status"
+)
+
+var tests = map[string]struct {
+	model string
+	code  int
+}{
+	//	"case1": {
+	//		// Parse error
+	//		model: `
+	//type user
+	//type self
+	//  relations
+	//    define member: [user] as self
+	//`,
+	//	},
+	//	"case2": {
+	//		// Parse error
+	//		model: `
+	//type user
+	//type this
+	//  relations
+	//    define member: [user] as self
+	//`,
+	//	},
+	//	"case3": {
+	//		// Parse error
+	//		model: `
+	//type user
+	//type group
+	//  relations
+	//    define self: [user] as self
+	//`,
+	//	},
+	//	"case4": {
+	// 		// Parse error
+	//		model: `
+	//type user
+	//type group
+	// relations
+	//   define this: [user] as self
+	//`,
+	//	},
+	//	"case5": {
+	//		// TODO: Can we validate this?
+	//		model: `
+	//type user
+	//type team
+	//  relations
+	//    define parent: [group] as self
+	//    define viewer as viewer from parent
+	//type group
+	//  relations
+	//    define parent: [team] as self
+	//    define viewer as viewer from parent
+	//`,
+	//	},
+	"case6": {
+		model: `
+type user
```

Comment:
Mind indenting these like others in the project? For example

https://github.com/openfga/openfga/blob/main/pkg/typesystem/typesystem_test.go#L1809-L1817

Reply:
I don't particularly like it, but okay.